onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -171,6 +171,90 @@ activation.)DOC")
       })
       .SetDoc(R"DOC(Tokenizer divides each string in X into a vector of strings along the last axis. All input strings including attributes are UTF-8 encoded.)DOC");
 
+  ONNX_CONTRIB_OPERATOR_SCHEMA(Ngram)
+      .SetDomain(kMSDomain)
+      .SinceVersion(1)
+      .Input(0, "X", "Input for n-gram extraction", "T")
+      .Output(0, "Y", "NGram results", "T1")
+      .TypeConstraint(
+          "T",
+          {"tensor(string)", "tensor(int32)", "tensor(int64)"},
+          "Input is ether string UTF-8 or int32/int64")
+      .TypeConstraint(
+          "T1",
+          {"tensor(float)"},
+          "1-D tensor of floats")
+      .Attr(
+          "N",
+          "Maximum n-gram length",
+          AttributeProto::INT)
+      .Attr(
+          "M",
```

Comment:
>M [](start = 11, length = 1)

Can we be a little less cryptic and use 'Min' and 'Max' as the attribute names? Attributes tend to have more meaningful names compared to inputs/outputs.

Reply:
Using longer name is clearer when user reads that attribute. However, when it comes to documentation of the whole operator's behavior, it's hard to have a clean description (similar things happen in mathematical papers where symbols are used a lot.).

Reply:
I'm thinking more of the user reading the attribute. Either way the documentation needs to fully explain it, but I would that thought that users are less likely to accidentally mix up what M and N are for if they are called 'min' and 'max'. 

---
In reply to: [243034207](https://github.com/Microsoft/onnxruntime/pull/180#discussion_r243034207) [](ancestors = 243034207)